### PR TITLE
Added clusterName to podSpec

### PR DESCRIPTION
**Pull request by jshaofuturewei**
Apply the following pod.json
{
"kind":"Pod",
"apiVersion":"v1",
"metadata":{
"name":"nginx",
"labels":{
"run":"nginx"
}
},
"spec":{
"containers":[
{
"name":"nginx1",
"image":"nginx:1.10.0",
"resources":{
"requests":{
"cpu":"100m"
}
}
}
]
}
}
After the pod is created, please run the following command to bind a cluster

curl -X POST http://127.0.0.1:8080/api/v1/namespaces/default/pods/nginx/binding -H "Content-Type: application/json" -d @binding.json

binding.json
{
"apiVersion":"v1",
"kind":"Binding",
"metadata": {
"name":"nginx"
},
"target": {
"apiVersion":"v1",
"kind":"Cluster",
"name":"c1"
}
}

{
"apiVersion":"v1",
"kind":"Binding",
"metadata": {
"name":"nginx"
},
"target": {
"apiVersion":"v1",
"kind":"Scheduler",
"name":"s1"
}
}

The result is
{
"kind": "Status",
"apiVersion": "v1",
"metadata": {

},
"status": "Success",
"code": 201
}

And then search pods by its clusterName
cluster/kubectl.sh get pods --field-selector spec.clusterName="c1"
Or by its assignedSchedulerName
cluster/kubectl.sh get pods --field-selector status.assignedScheduler.name="s1"
Or by its status
cluster/kubectl.sh get pods --field-selector status.phase="assigned" (for Scheduler)
cluster/kubectl.sh get pods --field-selector status.phase="binded" (for Cluster)

### CODE REVIEW

**Review by kimeunju108 — APPROVED** [review]
LGTM

File: pkg/registry/core/pod/storage/storage.go [hunk 1]
```diff
@@ -188,18 +188,29 @@ func (r *BindingREST) setPodHostAndAnnotations(ctx context.Context, podID, oldMa
...
 			return nil, fmt.Errorf("pod %s is being deleted, cannot be assigned to a host", pod.Name)
 		}
-		if pod.Spec.VirtualMachine == nil && pod.Spec.NodeName != oldMachine {
-			return nil, fmt.Errorf("pod %v is already assigned to node %q", pod.Name, pod.Spec.NodeName)
+		podCondition := api.PodScheduled
+
+		switch targetKind {
+		case "Scheduler":
+			pod.Status.AssignedScheduler.Name = targetName
+			podCondition = api.SchedulerAssigned
```
Comment (cathyhongzhang) [inline pkg/registry/core/pod/storage/storage.go:196]: change to assigned

File: pkg/registry/core/pod/storage/storage.go [hunk 2]
```diff
@@ -188,18 +188,29 @@ func (r *BindingREST) setPodHostAndAnnotations(ctx context.Context, podID, oldMa
...
+		case "Cluster":
+			pod.Spec.ClusterName = targetName
+			podCondition = api.ClusterBound
```
Comment (cathyhongzhang) [inline pkg/registry/core/pod/storage/storage.go:199]: change to binded

File: pkg/registry/core/pod/storage/storage.go (hunk 1 above)
Comment (jshaofuturewei) [reply pkg/registry/core/pod/storage/storage.go:196]: Updated

File: pkg/registry/core/pod/storage/storage.go (hunk 2 above)
Comment (jshaofuturewei) [reply pkg/registry/core/pod/storage/storage.go]: Updated in the same request

**Review by CoderKevinZhang — APPROVED** [review]
I added a comment in my dispatcher controller with "Start Dispatcher Process", please feel free to add your process starting method under this comment

**Pull request merged**